sdk/cosmosdb/cosmos/src/retry/RetryContext.ts

```diff
@@ -4,4 +4,5 @@ export interface RetryContext {
   retryCount: number;
   retryRequestOnPreferredLocations?: boolean;
   clearSessionTokenNotAvailable?: boolean;
+  retryLocationIndex?: number;
```

Comment:
It is unclear what this variable means. Please add detailed comments here.

Reply:
updated